### ISSUE TITLE
[FYI] A workaound for cv_bridge_boost's ImportError on Windows

### ISSUE THREAD

**Issue by sgr-aoba** [ros-perception/vision_opencv#537]
This is a workaround for cv_bridge_boost's ImportError with ROS2 on Windows (humble).
I hope this helps anyone encountering the same error.

# Environment

| Component | Version |     |
| --------- | ------- | --- |
| Operating System | Windows 10 22H2 |
| ROS2 Release | Humble | Binary installation by following [the guide](https://docs.ros.org/en/humble/Installation/Windows-Install-Binary.html). |
| vision_opencv | [Humble branch](https://github.com/ros-perception/vision_opencv/tree/humble) | Installed as a package (by git submodule).  And set the environment variable `OpenCV_DIR`. |
| boost | 1.74.0 (msvc-14.2-64) | Binary installation from [sourceforge](https://sourceforge.net/projects/boost/files/boost-binaries/1.74.0/). And set the environment variable `BOOST_ROOT`. |

# Steps to reproduce issue

- Create a python node using CvBridge
- Build (colcon build)
- Run

# Expected behavior

Nodes using CvBridge work as expected.

# Actual behavior

Nodes using CvBridge will exit with the following error.

```
ImportError: DLL load failed while importing cv_bridge_boost: The specified module could not be found.
```

# Workaround

It need to add DLL paths [explicitly](https://docs.python.org/3.8/library/os.html#os.add_dll_directory).
Add the following snippet before importing CvBridge.

```python
import platform
if platform.system() == 'Windows':
    import os
    from pathlib import Path
    from ament_index_python.packages import get_package_prefix
    dll_dir: Callable[[Path], Path] = lambda p: next(p.rglob('*.dll')).parent
    for d in [
        Path(os.environ.get('BOOST_ROOT')),
        Path(os.environ.get('OPENCV_DIR')),
        Path(get_package_prefix('cv_bridge')),
    ]:
        os.add_dll_directory(dll_dir(d))

from cv_bridge import CvBridge, CvBridgeError
```

**Comment by sgr-aoba** [comment]
Related issues: #247, ms-iot/ROSOnWindows#371

**Comment by AndreasGronlund** [comment]
Tried the above fix without success. I get the error message "ImportError: DLL load failed while importing cv_bridge_boost: The specified module could not be found"

**Comment by sgr-aoba** [comment]
Have you built cv_bridge?
ROS for Windows does not include cv_bridge.

**Comment by AndreasGronlund** [comment]
Yes, cv_bridge is downloaded and built (branch humble, as is the windows ros2 binary) with no errors from what I can see, but the module cannot be found. Cannot find cv_bridge_boost which should be imported on L96 (from cv_bridge.boost.cv_bridge_boost import getCvType) and L138 (from cv_bridge.boost.cv_bridge_boost import cvtColor2) in core.py (cv_bridge/core.py).

**Comment by sgr-aoba** [comment]
It seems that your problem is a issue with building of cv_bridge.
The introduction above assumes that cv_bridge is available as a package.

How about trying git clone ros-perception to `<YOUR_ROS2_PROJECT>/src`, and run `colcon build`.

**Comment by ijnek** [comment]
Thanks for posting this workaround for others! I'm going to close it as #247 tracks the actual issue.
